DynamicResourceInfo: Allow internal loading of BinaryResource from URL

jmiserez opened the pull request:
Add method to DynamicResourceInfo to support the loading of a
BinaryResource only by IUiSession, without knowing the IJsonAdapter
directly.

329612

fschinkel commented on org.eclipse.scout.rt.ui.html/src/main/java/org/eclipse/scout/rt/ui/html/res/loader/DynamicResourceInfo.java:
```diff
@@ -106,6 +108,31 @@ public static DynamicResourceInfo fromPath(HttpServletRequest req, String path)
+   * @param path
+   *          decoded path (non url-encoded)
+   */
+  public static DynamicResourceInfo fromPath(Function<String/*uiSessionId*/, IUiSession> uiSessionResolver, Function<String/*id*/, IJsonAdapter> jsonAdapterResolver, String path) {
```
`uiSessionResolver` is not really used. It is called and the result is validated (matching ids), but the resulting `IUiSession` is not used. Therefore, delete the parameter or use it to validate the `IJsonAdapter`.

jmiserez replied:
Fixed.

fschinkel commented on org.eclipse.scout.rt.ui.html/src/main/java/org/eclipse/scout/rt/ui/html/res/loader/DynamicResourceInfo.java:
```diff
@@ -106,6 +108,31 @@ public static DynamicResourceInfo fromPath(HttpServletRequest req, String path)
+    }
+    // resolve session id and adapter id
+    IUiSession uiSession = Optional.ofNullable(parts.getUiSessionId()).map(uiSessionResolver).orElse(null);
+    IJsonAdapter jsonAdapter = Optional.ofNullable(parts.getAdapterId()).map(jsonAdapterResolver).orElse(null);
```
call `jsonAdapterResolver` after validating the `uiSession`

jmiserez replied:
We pass in IUiSession directly now, removed resolver indirection.

fschinkel commented on org.eclipse.scout.rt.ui.html/src/main/java/org/eclipse/scout/rt/ui/html/res/loader/DynamicResourceLoader.java:
```diff
@@ -81,7 +81,34 @@ public BinaryResource loadResource(String pathInfo) {
+    return localResourceHolder.get();
+  }
+
+  protected static IBinaryResourceProvider getBinaryResourceProvider(IUiSession uiSession, String adapterId) {
```
Please do not change this to static on a released version as it might break existing projects.

jmiserez replied:
Reverted this change to DynamicResourceLoader.

jmiserez committed:
DynamicResourceInfo: Allow internal loading of BinaryResource from URL

Add method to DynamicResourceInfo to support the loading of a
BinaryResource only by IUiSession, without knowing the IJsonAdapter
directly.

329612

jmiserez commented:
Related PRs:
- 22.0 https://github.com/eclipse-scout/scout.rt/pull/956
- 23.2 https://github.com/eclipse-scout/scout.rt/pull/1183 (cherry-pick)
- 24.2 https://github.com/eclipse-scout/scout.rt/pull/1184 (cherry-pick)
- 25.1 https://github.com/eclipse-scout/scout.rt/pull/1185 (cherry-pick)
